### ISSUE TITLE
Does the Speculation Rules API support hash routing?

### ISSUE THREAD

**Issue by Rosadojonathan** [WICG/nav-speculation#314]
Hello, 

I've been trying to get the Speculation Rules API to work with my app. I was wondering if the API was supposed to work with hash routing?

Two of my URLs which use hash routing show the hash portion in a red font which leads me to believe it didn't work properly
![Screenshot 2024-05-07 at 19 49 26](https://github.com/WICG/nav-speculation/assets/25138966/b933327b-8b56-416b-b910-8ff49c349eed)
and when I navigate to those pages by clicking on the link, I don't seem to benefit from the prerendering.

Is there something I'm missing?

Thanks!

**Comment by tunetheweb** [comment]
I presume by "hash routing" you mean an SPA type architecture?

If so, then no, that is not supported by speculation rules: https://developer.chrome.com/docs/web-platform/prerender-pages#speculation_rules_and_spas

**Comment by Rosadojonathan** [comment]
Yes, indeed. So I should switch from prerendering to prefetching, and it should still help the loading performance to a certain extent. Thank you!

**Comment by tunetheweb** [comment]
I would test that to be sure, because Speculation Rules will fetch this as a document, rather than as a your SPA may fetch this (e.g. as a JSON XHR fetch request). You may be better to see what your SPA framework offers for prefetching rather than use Speculation Rules.

**Comment by domenic** [comment]
I wonder what a good DevTools UI would look like here, to signal that you performed a SPA navigation and so prefetch/prerender does not apply.

**Comment by domenic** [comment]
Issue triage: let me close this. https://issues.chromium.org/issues/40257463 is the Chromium-side bug to make this better behaved on misuse. If people have ideas for what we could do in DevTools to signal the problem, then another Chromium bug is probably the way to go.
